src/Essentials/src/FileSystem/FileSystem.ios.cs

```diff
@@ -158,68 +160,134 @@ protected override void Dispose(bool disposing)
 		}
 	}
 
-	class UIDocumentFileResult : FileResult
+	class UIDocumentFileResult : MediaFileResult
 	{
-		internal UIDocumentFileResult(NSUrl url)
-			: base()
+		UIDocument document;
+		NSUrl assetUrl;
+
+		internal UIDocumentFileResult(NSUrl assetUrl, string fileName)
 		{
-			var doc = new UIDocument(url);
-			FullPath = doc.FileUrl?.Path;
-			FileName = doc.LocalizedName ?? Path.GetFileName(FullPath);
+			this.assetUrl = assetUrl;
+			document = new UIDocument(assetUrl);
+			Extension = document.FileUrl.PathExtension!;
```

Comment:
re we sure we want to ignore the PathExtension being null? can Extension be already nullable?